src/pages/BlogArticle.tsx

```diff
@@ -1,3 +1,56 @@
-const BlogArticle = () => <>BlogArticle</>;
+import { Box, Button, Typography } from "@mui/material";
+import { useBlog } from "../services/queries/useBlog";
+import { formatDate } from "../utils/formatDate";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import { useNavigate } from "react-router-dom";
+
+const BlogArticle: React.FC = () => {
+  const { isLoading, blog, error } = useBlog();
+  const navigate = useNavigate();
+
+  if (isLoading) return <p>Loading...</p>;
+  if (error) return <p>Error: {error}</p>;
+  if (!blog) return <Typography>No blog found</Typography>;
+
+  const { title, image, fullArticle, createdAt } = blog;
+
+  const paragraphs =
```

Comment:
I would use memo here, what if there will be a long article? maybe if there will be no re-renders, than it's not necessary